src/targets/gpu/include/migraphx/gpu/group_query_attention.hpp

```diff
@@ -71,63 +64,53 @@ struct gqa_parameters
-               "MIGRAPHX_MAKE_CONSTANT(int32_t{" + std::to_string(local_window_size) + "}), " +
-               "MIGRAPHX_MAKE_CONSTANT(bool{" +
-               std::to_string(static_cast<int>(rotary_interleaved)) + "}), " +
-               "MIGRAPHX_MAKE_CONSTANT(bool{" +
-               std::to_string(static_cast<int>(past_present_share_buffer)) + "})";
+               "}), " + "MIGRAPHX_MAKE_CONSTANT(uint32_t{" + std::to_string(kv_num_heads) + "}), " +
+               "MIGRAPHX_MAKE_CONSTANT(bool{" + std::to_string(static_cast<int>(interleaved)) +
+               "})";
     }
 };
 
 static inline gqa_parameters init_params(const std::vector<shape>& inputs, const value& v)
 {
-    auto num_heads          = v.at("num_heads").to<std::uint32_t>();
-    auto kv_num_heads       = v.at("kv_num_heads").to<std::uint32_t>();
-    auto do_rotary          = v.at("do_rotary").to<bool>();
-    auto local_window_size  = v.at("local_window_size").to<std::uint32_t>();
-    auto rotary_interleaved = v.at("rotary_interleaved").to<bool>();
-    auto scale              = v.at("scale").to<float>();
-    auto present_kv_seqlen  = inputs[1].lens().size() == 4 ? inputs[1].lens()[2] : 0;
+    std::size_t num_heads = -1;
+    if(contains(v, "num_heads"))
+        num_heads = v.at("num_heads").to<std::uint32_t>();
+    auto kv_num_heads = v.at("kv_num_heads").to<std::uint32_t>();
+    auto interleaved  = false;
+    if(v.contains("interleaved"))
+        interleaved = v.at("interleaved").to<bool>();
+    auto present_kv_seqlen = inputs[1].lens().size() == 4 ? inputs[1].lens()[2] : 0;
 
     const auto& q_shape               = inputs[0];
     const auto& q_lens                = q_shape.lens();
     const std::size_t batch_size      = q_lens[0];
     const std::size_t sequence_length = q_lens[2];
-    std::size_t head_size             = q_lens[3];
-    auto q_hidden_size                = kv_num_heads * head_size;
+    const std::size_t head_size       = q_lens[3];
 
     std::size_t rotary_dim = inputs.size() >= 4 ? inputs[3].lens()[1] * 2 : 0;
     if(inputs.size() == 3)
     {
         present_kv_seqlen = inputs[2].lens()[2];
     }
-    auto seq_stride                = head_size;
-    auto head_stride               = sequence_length * seq_stride;
-    auto batch_stride              = (num_heads + 2 * kv_num_heads) * head_stride;
-    auto position_ids_format       = sequence_length == 1 ? 1 : 0;
-    bool past_present_share_buffer = true;
+    auto seq_stride          = head_size;
+    auto head_stride         = sequence_length * seq_stride;
+    auto batch_stride        = (num_heads + 2 * kv_num_heads) * head_stride;
+    auto position_ids_format = sequence_length == 1 ? 1 : 0;
+
     gqa_parameters gqa_params;
-    gqa_params.batch_size                = batch_size;
-    gqa_params.sequence_length           = sequence_length;
-    gqa_params.hidden_size               = q_hidden_size;
```

Comment:
Why remove hidden_size?

Reply:
Ah okay. I only ask as some flavors of attention do use that (its also called embeddng size) but they then reshape that dimension (batch, seq len, hidden_size) -> (batch, seq len, num_heads, head_size) before reshaping it to do calculation across heads. 

If its not needed here, ya remove it.